src/plugins/searchDecorations.js

```diff
@@ -15,10 +15,12 @@ export default function searchDecorations() {
 				const searchResults = runSearch(doc, '')
 				return highlightResults(doc, searchResults)
 			},
-			apply(transaction, _oldState) {
-				const query = transaction.getMeta('searchQuery') ?? ''
+			apply(transaction, oldState) {
+				const query = transaction.getMeta('searchQuery')
 
-				if (transaction.docChanged || query) {
+				if (query === undefined) {
+					return oldState
+				} else if (transaction.docChanged || query) {
```

Comment:
What will happen if `transaction.docChanged` is set and there is no `searchQuery` meta set?

Is that the intended behavior?

If you now highlight a word and then type this word into the doc - will it be highlighted?

I suspect not as the query did not change thus there was no searchQuery meta so `query` is undefined here and the state stays the same. But I might be wrong.

Reply:
You make a good point. Originally it was not like this, but then I was looking into the bug Julius described in his comment above, and determined that when a sync request comes in, the editor state is reset (or, the plugin state at least). So I was trying to find a way to keep the highlighting decorations in that case. But based on your logic it would have unintended effects elsewhere.

Will have to look into it further

Reply:
I think you will need the state of the `searchQuery` plugin so you can know what the current search query is even if it has not changed.
If the doc changed or the search query changed (including turning to '') you need to recompute the decoration with the current search query (the one from `searchQuery` plugin if it did not change).